core/src/main/java/bisq/core/trade/protocol/bsq_swap/tasks/seller/SellerCreatesAndSignsTx.java

```diff
@@ -66,29 +63,32 @@ protected void run() {
 
             // At first we try with min. tx size
             int sellersTxSize = MIN_SELLERS_TX_SIZE;
+            Coin sellersBtcChangeAmount = Coin.ZERO;
             Coin required = BsqSwapCalculation.getSellersBtcInputValue(trade, sellersTxSize, sellersTradeFee);
 
+            // We do one iteration here to get the size of the inputs (segwit or not)
+            inputsAndChange = tradeWalletService.getSellersBtcInputsForBsqSwapTx(required);
+            sellersTxSize = BsqSwapCalculation.getVBytesSize(inputsAndChange.first, 0);
+            required = BsqSwapCalculation.getSellersBtcInputValue(trade, sellersTxSize, sellersTradeFee);
```

Comment:
I don't understand why this is needed, seems it's covered in the loop

Reply:
Yes thats another complicate issue. First iteration we use the min tx size of 104 vbytes. if you have the exact inputs for the expected tx where input is a non-segwit input it would create a change output (diff of larger tx size for non segwit and min tx size). that would trigger then trigger a calculation for 1 nonsegwit input (correct) + 1 output (not correct) and would cause a InsufficientMoneyException. To avoid that I set the change at the frist iteration to 0 so we get the correct size for the input and then do the loop where we get then the correct result with 0 change and the exact funds are used up.

Will likely require more tests with diff. such scenarios. But at least the ones I tested work with the current code.